### PR TITLE
dynamic chain sync window

### PR DESCRIPTION
**Pull request by vyzo**
Additive Increase, Multiplicative Decrease -- TCP style.
It should help with chain sync in crappy networks and accelerate stuff in good ones.

### CODE REVIEW

File: chain/sync.go [hunk 1]
```diff
@@ -1476,8 +1486,16 @@ func (syncer *Syncer) iterFullTipsets(ctx context.Context, headers []*types.TipS
 			}
 		}
 		i -= batchSize
+
+		if i >= 0 {
+			windowSize += 10
+			log.Infof("successfully fetched %d messages; increasing window size to %d", len(bstout), windowSize)
+		}
```
Comment (magik6k) [inline chain/sync.go:1493]: Should make sure to not go over `blocksync.MaxRequestLength` (finality / 900 blocks)

**Review by magik6k — COMMENTED** [review]
1 Nit

File: chain/sync.go (hunk 1 above)
Comment (vyzo) [reply chain/sync.go]: ah, good point -- I wasn't aware of that limit! fixing.

Comment (vyzo) [reply chain/sync.go]: fixed

**Review by magik6k — APPROVED** [review]

**Pull request merged**